### DISCUSSION TITLE
looking for LSTM example

### DISCUSSION THREAD

**Discussion by AnouarITI** [fastmachinelearning/hls4ml#725] [Q&A]
Hello
I need to make some benchmarks of an LSTM on HLS4ML. I know that it is supported , but I did not find any concrete example. I found an already trained model in one repository but I need the training data also since i need also to prune the model and test its effects.

I would be very grateful for any help. Thank you

**Comment by jmduarte** [comment]
Hi @AnouarITI you should be able to find a small example of building and converting an LSTM in our tests: https://github.com/fastmachinelearning/hls4ml/blob/main/test/pytest/test_rnn.py

and this paper: https://arxiv.org/abs/2207.00559
@elhamekhoda may be able to give additional pointers to an example

**Reply by elhamekhoda** [reply]
Hi @AnouarITI Unfortunately, we did not add any LSTM or GRU example notebook to our `hls4ml tutorial` yet. I might be able to collect some example scripts used for the RNN paper studies. We will point you once we have an example ready.

You should be able to convert Keras `LSTM` or `GRU` layers out of the box using the latest `hls4ml`. Let us know if you have any specific issues while building your model with `hls4ml`
